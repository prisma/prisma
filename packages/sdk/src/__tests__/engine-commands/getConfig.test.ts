import stripAnsi from 'strip-ansi'
import { getConfig } from '../..'

describe('getConfig', () => {
  test('empty config', async () => {
    const config = await getConfig({
      datamodel: `
      datasource db {
        provider = "sqlite"
        url      = "file:../hello.db"
      }
      
      model A {
        id Int @id
        name String
      }`,
    })

    expect(config).toMatchSnapshot()
  })

  test('sqlite and createMany', async () => {
    expect.assertions(1)
    try {
      await getConfig({
        datamodel: `
      datasource db {
        provider = "sqlite"
        url      = "file:../hello.db"
      }

      generator client {
        provider = "prisma-client-js"
        previewFeatures = ["createMany"]
      }
      
      model A {
        id Int @id
        name String
      }`,
      })
    } catch (e) {
      expect(stripAnsi(e.message)).toMatchInlineSnapshot(`
        "Get config: Database provider \\"sqlite\\" and the preview feature \\"createMany\\" can't be used at the same time.
          Please either remove the \\"createMany\\" feature flag or use any other database type that Prisma supports: postgres, mysql or sqlserver."
      `)
    }
  })

  test('with generator and datasource', async () => {
    const config = await getConfig({
      datamodel: `
    datasource db {
      url = "file:dev.db"
      provider = "sqlite"
    }

    generator gen {
      provider = "fancy-provider"
      binaryTargets = ["native"]
    }

    model A {
      id Int @id
      name String
    }`,
    })

    expect(config).toMatchSnapshot()
  })

  test('datasource with env var', async () => {
    process.env.TEST_POSTGRES_URI_FOR_DATASOURCE =
      'postgres://user:password@something:5432/db'

    const config = await getConfig({
      datamodel: `
      datasource db {
        provider = "postgresql"
        url      = env("TEST_POSTGRES_URI_FOR_DATASOURCE")
      }
      `,
    })

    expect(config).toMatchSnapshot()
  })

  test('datasource with env var - ignoreEnvVarErrors', async () => {
    const config = await getConfig({
      ignoreEnvVarErrors: true,
      datamodel: `
      datasource db {
        provider = "postgresql"
        url      = env("SOMETHING-SOMETHING-1234")
      }
      `,
    })

    expect(config).toMatchSnapshot()
  })
})
