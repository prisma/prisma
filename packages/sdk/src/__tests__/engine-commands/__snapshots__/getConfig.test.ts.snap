// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getConfig datasource with env var - ignoreEnvVarErrors 1`] = `
Object {
  "datasources": Array [
    Object {
      "activeProvider": "postgresql",
      "name": "db",
      "provider": Array [
        "postgresql",
      ],
      "url": Object {
        "fromEnvVar": "SOMETHING-SOMETHING-1234",
        "value": null,
      },
    },
  ],
  "generators": Array [],
  "warnings": Array [],
}
`;

exports[`getConfig datasource with env var 1`] = `
Object {
  "datasources": Array [
    Object {
      "activeProvider": "postgresql",
      "name": "db",
      "provider": Array [
        "postgresql",
      ],
      "url": Object {
        "fromEnvVar": "TEST_POSTGRES_URI_FOR_DATASOURCE",
        "value": "postgres://user:password@something:5432/db",
      },
    },
  ],
  "generators": Array [],
  "warnings": Array [],
}
`;

exports[`getConfig empty config 1`] = `
Object {
  "datasources": Array [
    Object {
      "activeProvider": "sqlite",
      "name": "db",
      "provider": Array [
        "sqlite",
      ],
      "url": Object {
        "fromEnvVar": null,
        "value": "file:../hello.db",
      },
    },
  ],
  "generators": Array [],
  "warnings": Array [],
}
`;

exports[`getConfig with generator and datasource 1`] = `
Object {
  "datasources": Array [
    Object {
      "activeProvider": "sqlite",
      "name": "db",
      "provider": Array [
        "sqlite",
      ],
      "url": Object {
        "fromEnvVar": null,
        "value": "file:dev.db",
      },
    },
  ],
  "generators": Array [
    Object {
      "binaryTargets": Array [
        Object {
          "fromEnvVar": null,
          "value": "native",
        },
      ],
      "config": Object {},
      "name": "gen",
      "output": null,
      "previewFeatures": Array [],
      "provider": Object {
        "fromEnvVar": null,
        "value": "fancy-provider",
      },
    },
  ],
  "warnings": Array [],
}
`;
